### PR TITLE
Override opCodeForCorrespondingDirect API in OMR

### PR DESCRIPTION
**Pull request by liqunl**
Fixes: #4549

### CODE REVIEW

**Comment by liqunl** [comment]
@andrewcraik May I ask for your review?

**Comment by liqunl** [comment]
Depends on https://github.com/eclipse/omr/pull/5458

File: runtime/compiler/control/CompilationThread.cpp [hunk 1]
```diff
@@ -8471,8 +8471,9 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
                   }
 
                // See if we need to insert GCR trees
-               if (!details.supportsInvalidation() ||
-                   options->getOptLevel() >= hot) // Workaround for a bug with GCR inserted in hot bodies. See #4549 for details.
+               //if (!details.supportsInvalidation() ||
```
Comment (andrewcraik) [inline runtime/compiler/control/CompilationThread.cpp:8474]: This surprises me - did you intend to include this change with everything else it doesn't necessarily seem related??

Comment (liqunl) [reply runtime/compiler/control/CompilationThread.cpp:8474]: My bad, I accidentally committed it to this PR. I used this change to verify the fix in the reported issue. Will remove it.

**Comment by andrewcraik** [comment]
Jenkins test sanity all jdk15

**Comment by andrewcraik** [comment]
Jenkins test sanity all jdk15

**Comment by andrewcraik** [comment]
AIX is offline, but this is clean on all the other platforms. Travis failure was because the dependency hadn't merged when the last push happened. this should be good to go (esp since the API is not currently used).

**Pull request merged**